### DISCUSSION TITLE
Upgrading react-select V2 to V5 giving type error

### DISCUSSION THREAD

**Discussion by mertcihankurel** [JedWatson/react-select#4924] [Q&A]
I'm using react-select library's V2. I needed to upgrade it to latest version V5 to use its new features on accessibility.

After I upgraded this version, last few days I'm trying to handle this issue. I couldn't find any solution yet.

In **Async.d.ts** file;

   ```
 import { ReactElement, RefAttributes } from 'react';
    import Select from './Select';
    import { GroupBase } from './types';
    import useAsync from './useAsync';
    import type { AsyncProps } from './useAsync';
    export type { AsyncProps };
    declare type AsyncSelect = <Option = unknown, IsMulti extends boolean = false, Group extends GroupBase<Option> = GroupBase<Option>>(props: AsyncProps<Option, IsMulti, Group> & RefAttributes<Select<Option, IsMulti, Group>>) => ReactElement;
    declare const AsyncSelect: AsyncSelect;
    export { useAsync };
    export default AsyncSelect;
```
comes by default from the library.

When I try to run the server, I get this Type error.

```
Type error: '=' expected

/Users/meceka/Documents/reactselect/node_modules/react-select/dist/declarations/src/Async.d.ts
Type error: '=' expected.  TS1005


    3 | import { GroupBase } from './types';
    4 | import useAsync from './useAsync';
  > 5 | import type { AsyncProps } from './useAsync';
      |             ^
    6 | export type { AsyncProps };
    7 | declare type AsyncSelect = <Option = unknown, IsMulti extends boolean = false, Group extends GroupBase<Option> = GroupBase<Option>>(props: AsyncProps<Option, IsMulti, Group> & RefAttributes<Select<Option, IsMulti, Group>>) => ReactElement;
    8 | declare const AsyncSelect: AsyncSelect;
```
Typescript version:

   `"typescript": "^3.2.2"`
React-select version:

   `"react-select": "^5.2.1"`
Does anybody have an idea?

**Comment by Rall3n** [comment]
Hello @mcihadkurel,

What version of TypeScript are you using specifically? You only provided the version string from your `package.json` which in itself is not the real version installed.

With that said, have you tried a newer version of TypeScript (`^4.0.0`)?

**Reply by mertcihankurel** [reply]
Hi, @Rall3n thank you for your answer and also for reminding me how to ask questions here. I have typescript latest 4.4.4 version on my local, but the project itself uses 3.2.2. Last 1 week whatever I do, I couldn't fix this issue. Any approach I came with brought another problem for some reason.

**Reply by Rall3n** [reply]
@mcihadkurel, sorry for the late reply.

How are you importing the components of `react-select`?
Has your project been created with the help of `create-react-app`? If so, what version of `typescript-eslint` is being used?

**Reply by mertcihankurel** [reply]
Hi @Rall3n,

Thank you for getting back to me. We are using react-select components as attached below. 
Our project created with create-react-app and version of typescripts are `"@typescript-eslint/eslint-plugin": "^1.5.0",
    "@typescript-eslint/parser": "^1.5.0",`
İt's been 20 days I'm still trying to fix it. I'll be glad if you give me an idea.
<img width="432" alt="Screen Shot 2021-12-09 at 13 52 45" src="https://user-images.githubusercontent.com/65771498/145383599-5d6f52f9-b4dc-4bce-9b1c-37d6f18eb906.png">

**Reply by Rall3n** [reply]
@mcihadkurel Last but not least, what version of `react-scripts` is installed?

**Reply by mertcihankurel** [reply]
Hi @Rall3n,

We don't have `react-scripts` in our project. İs it necessary to upgrade `react-select` ?

**Reply by Rall3n** [reply]
@mcihadkurel If you don't have `react-scripts`, has your project even been generated through `create-react-app`?
Either way, I suggest you update your `@typescript-eslint` packes to the latest (or nearest supported) version.

**Reply by mertcihankurel** [reply]
Hi @Rall3n, it didn't solve the issue. Still remains the same. Do you have any other options in your mind?

**Comment by ebonow** [comment]
@mcihadkurel Perhaps creating a codesandbox would be more helpful for someone to look at and diagnose.
